File: js/src/interfaces/NamespaceChunk.ts
import { CustomDataLink } from './CustomDataLink';

/**
 * A chunk of namespace data.
 * Contains links and a reference to the previous chunk.
 */
export interface NamespaceChunk {
  /**
   * CID of the previous chunk, or null if this is the first chunk
   */
  prev: string | null;
  
  /**
   * Links contained in this chunk
   */
  links: CustomDataLink[];
}

File: Circles.Profiles.Models/Core/NamespaceChunk.cs
using System.Text.Json.Serialization;

namespace Circles.Profiles.Models.Core;

/// <summary>
/// One “page” of a namespace.
/// • <c>prev</c> – CID of the next-older chunk (or <c>null</c> for the tail)
/// • <c>links</c> – chronologically stored <see cref="CustomDataLink"/>s
/// </summary>
public sealed record NamespaceChunk
{
    [JsonPropertyName("@context")] public string Context { get; init; } = JsonLdMeta.NamespaceContext;

    [JsonPropertyName("@type")] public string Type { get; init; } = "NamespaceChunk";

    [JsonPropertyName("prev")] public string? Prev { get; init; }
    [JsonPropertyName("links")] public List<CustomDataLink> Links { get; init; } = new();
}